plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/resource/ResourceAction.java

```diff
@@ -515,40 +515,62 @@ public boolean isContentListAttribute() {
         return false;
     }
 
+    protected Map getImgMetadata(InputStream input) {
+        Map<String, String> map = new HashMap<>();
+        
+        try {
+            Metadata meta = ImageMetadataReader.readMetadata(input);
+            map = processMetaData(meta);
+        } catch (ImageProcessingException ex) {
+            logger.error("Error reading metadata from file " + this.getFileName() + " - message " + ex.getMessage());
+        } catch (IOException ioex) {
+            logger.error("Error reading inputStream", ioex);
+        }
+        return map;
+    }
+    
     protected Map getImgMetadata(File file) {
```

Comment:
This method looks like a duplicate of the InputStream based one. Does it make sense to just convert the file to an input stream and have one method with logic rather than two perfect duplicates?

Reply:
This is necessary because we only have the File whenthe resource is created; when editing an existing resource the the file is purposely hidden by the resource manager and we only have the inputStream available for IO operations